tests/upgrades/conftest.py

```diff
@@ -342,11 +341,20 @@ def pytest_collection_modifyitems(items, config):
                 depend_on_node_ids = [_get_test_node_id(f) for f in dependant_on_functions]
                 for depend_on_node_id in depend_on_node_ids:
                     if depend_on_node_id in pre_upgrade_failed_tests:
-                        log(
-                            f'Deselected (because of dependant test failed): {item.nodeid}',
-                            'INFO',
-                        )
-                        deselected_items.append(item)
-
-        config.hook.pytest_deselected(items=deselected_items)
-        items[:] = [item for item in items if item not in deselected_items]
+                        item.add_marker(
+                            pytest.mark.fail(
+                                reason=f'The dependant test {item.nodeid} failed!'))
+
+
+class DependentTestFailed(Exception):
+    """Raise when the dependent test fails"""
+    pass
+
+
+@pytest.fixture(autouse=True)
+def fail_xfail_test(request):
+    """Called after fail mark added on post_upgrade tests using above collection modification"""
+    fail = request.node.get_closest_marker('fail')
+    if fail:
+        reason = fail.kwargs.get('reason')
+        raise DependentTestFailed(reason)
```

Comment:
I think it would be better to call `pytest.fail` here, not end the test in an ERROR during setup.

Reply:
As said, pytest doesn't allow me to fail the test from the test fixture. Even if I try to call `pytest.fail()`, pytest shows the test as error :
`E      Failed: The dependant test tests/testing/test_ups.py::TestSomethingWorks::test_postthing failed!`

Reply:
But raising an explicit exception provides a better granularity than just a plain `failed` error, and it would also help in ibutsu to filter tests with `DependentTestFailed` error!

So I would better stick to the explicit error!